### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by cesarvarela**
This should be merged after #247

### CODE REVIEW

**Comment by smcgregor** [comment]
It looks like this needs master merged in? Is this still in `draft` form?

**Comment by cesarvarela** [comment]
Updated.

It is ready, but I left it as `draft` because it depends on #247 to be merged first.

**Comment by smcgregor** [comment]
I merged staging to master and squashed in the process because the commit history had become messy. I then force updated the staging branch to match master. You will need to merge staging into this PR and then we can move it into staging.

**Comment by smcgregor** [comment]
@cesarvarela another package lock conflict. I'll merge this to staging after you update the package lock and after the current staging lands.

File: site/gatsby-site/src/pages/forgotpassword.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+
+const ForgotPasswordSchema = Yup.object().shape({
+  email: Yup.string().email('Invalid email').required('Required'),
+});
+
+const ForgotPassword = (props) => {
+  const {
+    actions: { sendResetPasswordEmail },
```
Comment (smcgregor) [inline site/gatsby-site/src/pages/forgotpassword.js:15]: I don't see where `sendResetPasswordEmail` is defined for this script.

File: site/realm/functions/onNewUser.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+exports = async (authEvent) => {
+
+  const mongodb = context.services.get("mongodb-atlas");
+  const users = mongodb.db("customData").collection("users");
```
Comment (smcgregor) [inline site/realm/functions/onNewUser.js:4]: Realm creates this `customData` collection? Does this PR require MongoDB changes?

File: site/realm/auth/providers.json [hunk 3]
```diff
@@ -14,8 +14,8 @@
         "type": "local-userpass",
         "config": {
             "autoConfirm": true,
-            "resetPasswordSubject": "Password Resets are not Currently Supported",
-            "resetPasswordUrl": "http://aiid.partnershiponai.org/",
+            "resetPasswordSubject": "",
+            "resetPasswordUrl": "http://localhost:8000/resetpassword",
```
Comment (smcgregor) [inline site/realm/auth/providers.json:18]: This needs to be updated to `https://incidentdatabase.ai/`?

**Comment by smcgregor** [comment]
* I have a couple inline questions.
* package-lock needs to be updated one more time.
* after this merges I am going to test it out in staging by editing the CSET classifications from a new admin account. If all goes well, we'll merge to production. The next steps for this functionality will be to add incident editor and taxonomy editor roles. The incident editors will be permitted to update incident reports and add incidents, while the taxonomy editors will be permitted to edit taxonomy classifications.

**Comment by cesarvarela** [comment]
Updated the reset password url and merged staging.

**Pull request merged**